### DISCUSSION TITLE
Is there a way to configure viewer.js without editing its code?

### DISCUSSION THREAD

**Discussion by chan-dev** [stephanrauh/ngx-extended-pdf-viewer#904] [Q&A] [answered]
I'm having an issue where adding a `file` query parameter throws an error even if i don't use the value of the query parameter as the pdf source.

![pdf-viewer-error](https://user-images.githubusercontent.com/46891115/132144856-44713fa4-c57f-422b-aa16-23b6c41855b8.png)

For instance, here's a sample URL
`chrome-extension://extensionId/src/pdfviewer/web/viewer.html?file=http://localhost:3000/apis/pdfstorage/someRandomId`

NOTE:
* it is just named `viewer.html` but it's our own custom html file

Here's how i actually use the library:
I just manually add the value in `[src]` `@Input`
```
<ngx-extended-pdf-viewer
  [src]="'assets/ws-checklist.pdf'"
  [handTool]="false"
  [showFindButton]="true"
  [useBrowserLocale]="true"
  zoom="100%"
  height="100vh"
></ngx-extended-pdf-viewer>
```

Upon searching the internet, it seems like they're all pointing to editing the source file. 
One example is from this stackoverflow [answer](https://stackoverflow.com/a/37379368/9732932)

Essentially, i have a couple of questions:
1. Does pdf.js looks at `?file=value` query parameter even if you don't use it in your code?
2. Is the only solution right now is to edit the viewer.js?

**Comment by stephanrauh** [comment, marked as answer]
Yes, you're right. I can't see how anybody benefits from the `?file=` URL parameter, so I should remove it. This is the offending line:

https://github.com/stephanrauh/pdf.js/blob/7c3b7214e5531b428c4c58593a6247d5bb8983bd/web/app.js#L2233

Would you mind to open a ticket?

**Comment by chan-dev** [comment]
I've already converted this into a new issue ticket. Thanks a lot.
On a second thought, is there ever a use case for `file` query parameter when using this library? I know it's useful for using the raw pdf.js file.

**Reply by stephanrauh** [reply]
It may also be put to good use in an Angular application, but I'm afraid most developers don't expect such a feature, so it only confuses them. It'd confuse me, FWIW. The default approach of Angular is using a router, so I assume most developer prefer to do just that.
